models/feedback.go

```diff
@@ -0,0 +1,55 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"net/mail"
+	"net/url"
+
+	"github.com/ONSdigital/dp-feedback-api/config"
+)
+
+type Feedback struct {
+	IsPageUseful      *bool  `json:"is_page_useful"`
+	IsGeneralFeedback *bool  `json:"is_general_feedback"`
+	OnsURL            string `json:"ons_url,omitempty"`
+	Feedback          string `json:"feedback,omitempty"`
+	Name              string `json:"name,omitempty"`
+	EmailAddress      string `json:"email_address,omitempty"`
+}
+
+func (f *Feedback) Validate(cfg *config.Config) error {
```

Comment:
I think it'd be neater to do the validation at the struct level like this:

https://github.com/go-playground/validator/blob/master/_examples/struct-level/main.go

Reply:
done - had to create a custom validator for ons url